File: python/6.py
#!/usr/bin/env python
def main(target):
    print(square_of_sums(target) - sum_of_squares(target))

def square_of_sums(target):
    sums = sum(range(1, target + 1))
    return sums ** 2

def sum_of_squares(target):
    squares = []
    for i in range(1, target + 1):
        squares.append(i ** 2)
    return sum(squares)

if __name__ == '__main__':
    main(100)

File: rust/src/bin/6.rs
fn main() {
    let target: i32 = 100;

    println!("{}", (square_of_sums(target) - sum_of_squares(target)));
}

fn square_of_sums(target: i32) -> i32 {
    let mut sum: i32 = 0;
    for i in 1..target+1 {
        sum += i;
    }
    sum * sum
}

fn sum_of_squares(target: i32) -> i32 {
    let mut sum: i32 = 0;
    for i in 1..(target + 1) {
        sum += i.pow(2);
    }
    sum
}
